packages/opentelemetry-cloud-trace-exporter/src/types.ts

```diff
@@ -133,8 +136,18 @@ export enum LinkType {
   PARENT_LINKED_SPAN = 2,
 }
 
-export interface SpansWithCredentials {
+/**
+ * A protobuf boolean
+ */
+export interface BoolValue {
+  value: boolean;
+}
+
+export interface NamedSpans {
```

Comment:
I would expect there is a way to generate TS stubs from the protobuf to avoid having to manually make these types, and also the ones above like `Span`.

Looking through the available official grpc/proto stuff, its a bit unclear which library to use. It looks like there are 3 official implementations:

1. [grpc](https://github.com/grpc/grpc-node/blob/master/PACKAGE-COMPARISON.md)
    - deprecated
    - looks like this is the one in this PR
2. [@grpc/grpc-js](https://github.com/grpc/grpc-node/blob/master/PACKAGE-COMPARISON.md) - looks like the new version of #1, pure javascript
3. [grpc-web](https://github.com/grpc/grpc-web)
    - also pure javascript
    - can generate typescript
    - intended for the browser, would it work in node?

Then there are a few other OSS libraries that can generate TS
- https://github.com/improbable-eng/ts-protoc-gen
- https://github.com/protobufjs/protobuf.js#usage-with-typescript

Thoughts?

Reply:
This is great feedback! 

1 (grpc) and 2 (@grpc/grpc-js) are functionally the same. In fact, you can simply change the imports and the code will work. The reason I chose the grpc package is that its compatible with node >=4, while @grpc/grpc-js requires node 8 or 12. I thought since the trace exporter's package.json says it requires node >=8 that the @grpc/grpc-js package had a narrower range then that.

grpc-web is specifically to make rpc calls from the browser. We cannot really use that because it requires a proxy server to convert request and response types between simple HTTP requests and rpc calls. 

Its definitely possible to use the TS stubs. In my experimentation I did actually implement the fully typed version using the protocol buffer compiler with the ts-protoc-gen plugin. I'd be happy to make this switch if people agree this is the way to go. One of the challenges I encountered with using the generated stubs is that it would require me to commit a fair number of files to the repo (~20 files: the proto files for google apis, the google common types they import, and their generated js and d.ts files).

I stuck with this implementation to start with because its consistent with what is being used in the OpenTelemetry [CollectorExporter](https://github.com/open-telemetry/opentelemetry-js/blob/bf2f6177eaaae416686c883c9efb076e862f29e9/packages/opentelemetry-exporter-collector/src/platform/node/CollectorExporter.ts#L79-L102) and this change is also quite minimal in comparison to the change required to use the generated stubs.

I haven't worked with protobufjs yet so I'll take a look!

Reply:
Cool that all makes sense, sounds like this PR is a good compromise as is